suite/tests/client-interface/cleancall.dll.c

```diff
@@ -41,6 +41,76 @@
 static reg_t buf[] = { 0xcafebabe, 0xfeedadad, 0xeeeeeeee, 0xbadcabee };
 #endif
 
+#ifdef X86
+/* buffers for testing reg_set_value_ex */
+byte orig_reg_val_buf[64];
+byte new_reg_val_buf[64];
+
+static void set_xmm()
+{
+    void *drcontext = dr_get_current_drcontext();
+    dr_mcontext_t mcontext = { sizeof(mcontext), DR_MC_ALL, };
+    dr_get_mcontext(drcontext, &mcontext);
+    reg_get_value_ex(DR_REG_XMM0, &mcontext, orig_reg_val_buf);
+    reg_get_value_ex(DR_REG_XMM0, &mcontext, new_reg_val_buf);
+    new_reg_val_buf[0] = 0x77;
+    new_reg_val_buf[2] = 0x89;
+    new_reg_val_buf[14] = 0x21;
+    bool succ = reg_set_value_ex(DR_REG_XMM0, &mcontext, new_reg_val_buf, 16);
+    ASSERT(succ == true);
+    dr_set_mcontext(drcontext, &mcontext);
+}
+
+static void check_xmm()
+{
+    void *drcontext = dr_get_current_drcontext();
+    dr_mcontext_t mcontext = { sizeof(mcontext), DR_MC_ALL, };
+    dr_get_mcontext(drcontext, &mcontext);
+    reg_get_value_ex(DR_REG_XMM0, &mcontext, new_reg_val_buf);
+    ASSERT(new_reg_val_buf[0] == 0x77);
```

Comment:
Don't use ASSERT for regular checks. Use an if () {} and stderr print an error. The expect/template should check for correctness. That said, you verified the test fails if value is wrong? Thanks for the test!

Reply:
Okay, I understood and will make the mods. Yes, I checked that the test fails if the value is incorrect, but I will also include a negation test which compares it with the original value.